### PR TITLE
Auto run unit tests on pull requests

### PR DESCRIPTION
**Pull request by steven4354**

### CODE REVIEW

File: .github/rust.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Rust
```
Comment (bigPingRail) [inline .github/rust.yml:1]: Change workflow name to "Rust unit tests" mb?

File: .github/rust.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ master, withdraw_from_anchor_if_negative_profit, run_unit_tests ]
```
Comment (bigPingRail) [inline .github/rust.yml:7]: withdraw_from_anchor_if_negative_profit, run_unit_tests is a temporary branches, right?
maybe we should leave only one branch on PR - master?

or, for example, do something like this
```
on:
  push:
    branches-ignore: 
      - "master"
  pull_request:
    branches: 
      - "master"
```

File: .github/rust.yml (hunk 1 above)
Comment (pronvis) [reply .github/rust.yml:1]: why `Rust ...` ?
I think better to name it `bAsset vault unit tests`

File: .github/rust.yml (hunk 2 above)
Comment (steven4354) [reply .github/rust.yml:7]: so I was thinking its best to run this on most situations to catch errors when they occur

Comment (bigPingRail) [reply .github/rust.yml]: yep this configuration allows to run tests in any other branches on push event except master, and run tests only on PR on master branch

**Pull request merged**